https://invent.kde.org/frameworks/plasma-framework/-/commit/a70957d96c29aa6059753cbfe23d0615ea688b36 and others broke compilation, as CMake's FindOpenGL is broken and assumes packages we do not have and shouldn't be needed either.
Let's make sure OpenGL::EGL is only used if it's actually found.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7da172546..e4d616a66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,11 +121,9 @@ add_feature_info(EGL OpenGL_EGL_FOUND
                  "A platform-agnostic mechanism for creating rendering surfaces for use with other graphics libraries, such as OpenGL|ES and OpenVG.")
 # OpenGL_EGL_FOUND is defined by FindOpenGL
 if(TARGET OpenGL::EGL)
-    set(HAVE_EGL ${OpenGL_EGL_FOUND})
-    set(EGL_TARGET OpenGL::EGL)
+set(HAVE_EGL 1)
 elseif(EGL_FOUND)
-    set(HAVE_EGL ${EGL_FOUND})
-    set(EGL_TARGET EGL::EGL)
+    set(HAVE_EGL 0)
     message(STATUS "Switch to EGL compatibility target EGL::EGL because OpenGL::EGL is not available")
 endif()
 
diff --git a/src/declarativeimports/core/CMakeLists.txt b/src/declarativeimports/core/CMakeLists.txt
index f35490ccb..7c58d1d47 100644
--- a/src/declarativeimports/core/CMakeLists.txt
+++ b/src/declarativeimports/core/CMakeLists.txt
@@ -59,7 +59,7 @@ if(HAVE_X11)
 endif()
 
 if(HAVE_EGL)
-    target_link_libraries(corebindingsplugin ${EGL_TARGET})
+target_link_libraries(corebindingsplugin OpenGL::EGL)
 else()
 
 endif()
